packages/std/src/mock.rs

```diff
@@ -1249,6 +1279,158 @@ mod tests {
         assert_eq!(dels, Some(del2c));
     }
 
+    #[test]
+    fn wasm_querier_works() {
+        let mut querier = WasmQuerier::default();
+
+        let any_addr = "foo".to_string();
+
+        // Query WasmQuery::Raw
+        let system_err = querier
+            .query(&WasmQuery::Raw {
+                contract_addr: any_addr.clone(),
+                key: b"the key".into(),
+            })
+            .unwrap_err();
+        match system_err {
+            SystemError::NoSuchContract { addr } => assert_eq!(addr, any_addr),
+            err => panic!("Unexpected error: {:?}", err),
+        }
+
+        // Query WasmQuery::Smart
+        let system_err = querier
+            .query(&WasmQuery::Smart {
+                contract_addr: any_addr.clone(),
+                msg: b"{}".into(),
+            })
+            .unwrap_err();
+        match system_err {
+            SystemError::NoSuchContract { addr } => assert_eq!(addr, any_addr),
+            err => panic!("Unexpected error: {:?}", err),
+        }
+
+        // Query WasmQuery::ContractInfo
+        let system_err = querier
+            .query(&WasmQuery::ContractInfo {
+                contract_addr: any_addr.clone(),
+            })
+            .unwrap_err();
+        match system_err {
+            SystemError::NoSuchContract { addr } => assert_eq!(addr, any_addr),
+            err => panic!("Unexpected error: {:?}", err),
+        }
+
+        querier.update_handler(|request| {
```

Comment:
I think the problem with this approach is, this is still a lot of custom code to provide. Just to be able to set and find some values in the mock storage.

Isn't there a better / more generic way to do that for these mocks?